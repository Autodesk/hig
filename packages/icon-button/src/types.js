export default Object.freeze({
  PRIMARY: "primary",
  FLAT: "flat",
  TRANSPARENT: "transparent"
});