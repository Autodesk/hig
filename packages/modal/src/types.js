export const types = Object.freeze({
  STANDARD: "standard",
  ALTERNATE: "alternate"
});

export const AVAILABLE_TYPES = Object.freeze(Object.values(types));
