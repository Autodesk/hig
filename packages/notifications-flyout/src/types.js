export const types = Object.freeze({
  ERROR: "error",
  PRIMARY: "primary",
  SUCCESS: "success",
  WARNING: "warning",
});

export const AVAILABLE_TYPES = Object.freeze(Object.values(types));
