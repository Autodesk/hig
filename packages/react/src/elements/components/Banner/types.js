/**
 * @todo Move into shared package to consolidate component types
 */

/** @type {Object.<string, string>} */
export const types = Object.freeze({
  PRIMARY: "primary",
  SUCCESS: "success",
  ERROR: "error",
  WARNING: "warning"
});

/** @type {string[]} */
export const AVAILABLE_TYPES = Object.freeze(Object.values(types));
