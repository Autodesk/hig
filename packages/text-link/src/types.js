/** @type {Object.<string, string>} */
export const types = Object.freeze({
  PRIMARY: "primary",
  SECONDARY: "secondary",
});

/** @type {string[]} */
export const AVAILABLE_TYPES = Object.freeze(Object.values(types));
