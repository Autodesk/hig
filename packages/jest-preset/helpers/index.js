/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
