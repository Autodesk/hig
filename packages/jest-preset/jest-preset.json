{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
