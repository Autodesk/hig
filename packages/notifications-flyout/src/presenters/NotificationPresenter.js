import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import RichText from "@weave-design/rich-text";
import ThemeContext from "@weave-design/theme-context";
import { createCustomClassNames } from "@weave-design/utils";

import DismissButtonPresenter from "./DismissButtonPresenter";
import stylesheet from "./stylesheet";

export default function NotificationPresenter(props) {
  const {
    children,
    dismissButtonTitle,
    hasHover,
    height,
    image,
    innerRef,
    onDismissButtonClick,
    onNotificationClick,
    onMouseEnter,
    onMouseLeave,
    showDismissButton,
    stylesheet: customStylesheet,
    timestamp,
    ...otherProps
  } = props;
  const { className } = otherProps;
  const notificationContentClassName = createCustomClassNames(
    className,
    "notification-content"
  );
  const notificationContentImageClassName = createCustomClassNames(
    className,
    "notification-content-image"
  );
  const notificationContentTextClassName = createCustomClassNames(
    className,
    "notification-content-text"
  );

  return (
    /* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(props, resolvedRoles);
        return (
          <div
            className={cx([className, css(styles.notification)])}
            onClick={onNotificationClick}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            ref={innerRef}
            role="listitem"
            style={{ height }}
          >
            <div
              className={cx([
                notificationContentClassName,
                css(styles.notificationContent),
              ])}
            >
              {image ? (
                <div
                  className={cx([
                    notificationContentImageClassName,
                    css(styles.notificationContentImage),
                  ])}
                >
                  {image}
                </div>
              ) : null}
              <div
                className={cx([
                  notificationContentTextClassName,
                  css(styles.notificationContentText),
                ])}
              >
                <RichText size="small">{children}</RichText>
                {timestamp}
                {showDismissButton ? (
                  <DismissButtonPresenter
                    className={className}
                    hasHover={hasHover}
                    onClick={onDismissButtonClick}
                    stylesheet={customStylesheet}
                    title={dismissButtonTitle}
                  />
                ) : null}
              </div>
            </div>
          </div>
        );
      }}
    </ThemeContext.Consumer>
    /* eslint-enable jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */
  );
}

NotificationPresenter.propTypes = {
  children: PropTypes.node,
  dismissButtonTitle: PropTypes.string,
  hasHover: PropTypes.bool,
  height: PropTypes.string,
  image: PropTypes.node,
  innerRef: PropTypes.func,
  onDismissButtonClick: PropTypes.func,
  onNotificationClick: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  showDismissButton: PropTypes.bool,
  stylesheet: PropTypes.func,
  timestamp: PropTypes.node,
};
