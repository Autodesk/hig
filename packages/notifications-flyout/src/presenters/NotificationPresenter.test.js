import { ENTERED } from "react-transition-group/Transition";
import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";

import NotificationPresenter from "./NotificationPresenter";
import { types } from "../types";

describe("notifications-flyout/presenters/NotificationPresenter", () => {
  takeSnapshotsOf(NotificationPresenter, [
    {
      desc: "renders without props",
      props: {},
    },
    {
      desc: "renders with all props",
      props: {
        children: "foobar",
        dismissButtonTitle: "hello world",
        featured: true,
        height: "3000px",
        innerRef: () => {},
        onDismissButtonClick: () => {},
        showDismissButton: true,
        timestamp: "2018-08-17",
        transitionStatus: ENTERED,
        type: types.SUCCESS,
        unread: true,
      },
    },
  ]);
});
