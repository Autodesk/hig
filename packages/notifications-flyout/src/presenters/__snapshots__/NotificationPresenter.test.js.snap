// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications-flyout/presenters/NotificationPresenter renders with all props 1`] = `
<div
  className="hig__notification-v1"
  role="listitem"
  style={
    Object {
      "height": "3000px",
    }
  }
>
  <div
    className="hig__notification-v1__content hig__notification-v1__content--success hig__notification-v1__content--featured hig__notification-v1__content--unread"
  >
    <div
      className="hig__notification-v1__content-text"
    >
      <div
        className="hig__rich-text hig__rich-text--small"
      >
        foobar
      </div>
      2018-08-17
      <div
        className="hig__notification-v1__dismiss-button"
      >
        <button
          className="hig__icon-button hig__icon-button--transparent"
          onClick={[Function]}
          title="hello world"
        >
          <svg
            className="hig__icon-button__icon"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.4 5.4l-.8-.8L8 7.3 5.4 4.6l-.8.8L7.3 8l-2.7 2.6.8.8L8 8.7l2.6 2.7.8-.8L8.7 8z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`notifications-flyout/presenters/NotificationPresenter renders without props 1`] = `
<div
  className="hig__notification-v1"
  role="listitem"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <div
    className="hig__notification-v1__content"
  >
    <div
      className="hig__notification-v1__content-text"
    >
      <div
        className="hig__rich-text hig__rich-text--small"
      />
    </div>
  </div>
</div>
`;
